libbeat/outputs/elasticsearch/config_otel.go

```diff
@@ -0,0 +1,179 @@
+		Hosts    []string `config:"hosts"  validate:"required"`
+	}{}
+
+	if err := esRawCfg.Unpack(&esToOTelOptions); err != nil {
+		return nil, fmt.Errorf("cannot parse Elasticsearch config: %w", err)
+	}
+
+	hosts := []string{}
+	for _, h := range esToOTelOptions.Hosts {
+		esURL, err := common.MakeURL(escfg.Protocol, escfg.Path, h, 9200)
+		if err != nil {
+			return nil, fmt.Errorf("cannot generate ES URL from host %w", err)
+		}
+		hosts = append(hosts, esURL)
+	}
+
+	// The workers config  can be configured using two keys, so we leverage
+	// the already existing code to handle it by using `output.HostWorkerCfg`.
+	workersCfg := outputs.HostWorkerCfg{}
+	if err := esRawCfg.Unpack(&workersCfg); err != nil {
+		return nil, fmt.Errorf("cannot read worker/workers from Elasticsearch config: %w", err)
+	}
+
+	headers := make(map[string]configopaque.String, len(escfg.Headers))
+	for k, v := range escfg.Headers {
+		headers[k] = configopaque.String(v)
+	}
+
+	otelTLSConfg, err := oteltranslate.TLSCommonToOTel(escfg.Transport.TLS)
+	if err != nil {
+		return nil, fmt.Errorf("cannot convert SSL config into OTel: %w", err)
+	}
+
+	otelYAMLCfg := map[string]any{
+		"logs_index":  esToOTelOptions.Index,    // index
+		"pipeline":    esToOTelOptions.Pipeline, // pipeline
+		"endpoints":   hosts,                    // hosts, protocol, path, port
+		"num_workers": workersCfg.NumWorkers(),  // worker/workers
+
+		// Authentication
+		"user":     escfg.Username, // username
+		"password": escfg.Password, // password
+		"api_key":  escfg.APIKey,   // api_key
+
+		// ClientConfig
+		"proxy_url":         esToOTelOptions.ProxyURL,         // proxy_url
+		"headers":           headers,                          // headers
+		"timeout":           escfg.Transport.Timeout,          // timeout
+		"idle_conn_timeout": &escfg.Transport.IdleConnTimeout, // idle_connection_connection_timeout
+		"tls":               otelTLSConfg,                     // tls config
+
+		// Retry
+		"retry": map[string]any{
+			"enabled":          true,
+			"initial_interval": escfg.Backoff.Init, // backoff.init
+			"max_interval":     escfg.Backoff.Max,  // backoff.max
+		},
+
+		// Batcher is experimental and by not setting it, we are using the exporter's default batching mechanism
+		// "batcher": map[string]any{
+		// 	"enabled":        true,
+		// 	"max_size_items": escfg.BulkMaxSize, // bulk_max_size
+		// },
+	}
+
+	// For type safety check only
+	// the returned valued should match `elasticsearchexporter.Config` type.
+	// it throws an error if non existing key names  are set
+	var result elasticsearchexporter.Config
+	d, _ := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
+		Squash:      true,
+		Result:      &result,
+		ErrorUnused: true,
+	})
+
+	err = d.Decode(otelYAMLCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO:
+	// // validates all required fields are set
+	// err = result.Validate()
+	// if err != nil {
+	// 	return nil, err
+	// }
```

Comment:
Is this supposed to be addressed before merging?

Reply:
so `Validate` required `compression` field to be [set](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/elasticsearchexporter/config.go#L277) and we did not have a matching field in beats configuration. 

I have now added `compression:gzip` (default value) field on our map structure to be able to use `Validate` method correctly. We can take the decision on how to map this correctly with our beats config in follow up PR's?